### PR TITLE
Oh the horrors! [MODULAR] [READY]

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request
Adds an ability to changelings called the horror form. It's basically them dropping their flesh disguise for their true form, a horrifying looking thing from the thing.

Changeling adaption stats:
	chemical_cost = 50
	dna_cost = 4//Tier 4
	req_dna = 15
	req_human = 1
Req absorbs = 1
The true changeling has a decent attack rate and can vent crawl. It is however quite slow. 

As a true changeling you have the ability to devour people which regenerates your health and damages them, making a lot of mess and noise!

## Why It's Good For The Game

These are genuinely quite fearsome and you should be scared of them. Adds a slight fear factor to the game.

## Known Issues
- Upon death of a true_changeling that is a changeling, a runtime will be thrown. This needs fixed, however it's not gamebreaking and can stay as it is for now.

- Changelings do not regenerate chemicals when dead.

## UPDATE
You now require a FULL absorb to actually buy this, and you require 15 DNA extracts. Also this fixes changelings not updating when dead. 
Chem cost is now 50
You can transform while in cuffs
You break cuffs when you transform(duh)
You will now regen chems when dead
It only deals 20 damage per attack however attacks at .5 instead of .7
You are only paralyzed for 10 seconds after death.

## Changelog
:cl:
add: Added a new ability for changelings; horror form
/:cl:

### CODE REVIEW

**Comment by SpookyTheFox** [comment]
NOPE NOPE NOPE.
I watched a test round, this is a horribly unbalanced antag. It gibbs people, breaks machines left and right, moves much faster than normal crew, can vent crawl (which lets it transit the station extremely fast), and revives faster than it can be gibbed by the shuttle, even with people constantly beating on it. For all intents and purposes, unkillable and free to murderbone with zero recourse.

It will have to loose ventcrawl at a bare minimum, as well as most of it's damage output, and get a substantial delay in revive time, as 8 seconds is much too short before this could be considered for use.

**Comment by Gandalf2k15** [comment]
> NOPE NOPE NOPE.
> I watched a test round, this is a horribly unbalanced antag. It gibbs people, breaks machines left and right, moves much faster than normal crew, can vent crawl (which lets it transit the station extremely fast), and revives faster than it can be gibbed by the shuttle, even with people constantly beating on it. For all intents and purposes, unkillable and free to murderbone with zero recourse.
> 
> It will have to loose ventcrawl at a bare minimum, as well as most of it's damage output, and get a substantial delay in revive time, as 8 seconds is much too short before this could be considered for use.

Currently has a speed of 2, which is almost the slowest mob we have.  Ventcrawl is it's movement and fear factor. Revives were bugged due to Life() not firing. Requires more of a group effort to kill than just "unga bunga assistant with toolbox." With a 2 move speed, anyone can run away. If you had died to it, then you were hunting it. That simple.

**Comment by chefferz** [comment]
> 
> 
> > NOPE NOPE NOPE.
> > I watched a test round, this is a horribly unbalanced antag. It gibbs people, breaks machines left and right, moves much faster than normal crew, can vent crawl (which lets it transit the station extremely fast), and revives faster than it can be gibbed by the shuttle, even with people constantly beating on it. For all intents and purposes, unkillable and free to murderbone with zero recourse.
> > It will have to loose ventcrawl at a bare minimum, as well as most of it's damage output, and get a substantial delay in revive time, as 8 seconds is much too short before this could be considered for use.
> 
> Currently has a speed of 2, which is almost the slowest mob we have. Ventcrawl is it's movement and fear factor. Revives were bugged due to Life() not firing. Requires more of a group effort to kill than just "unga bunga assistant with toolbox." With a 2 move speed, anyone can run away. If you had died to it, then you were hunting it. That simple.

Six security officers were unable to keep it contained. I watched as the HoS literally killed the ling, put it in the shuttle area to gib it, then it revived not even 8 seconds later to get away, despite missing literally every single organ and part. [https://i.imgur.com/gbEszO5.png](url)

**Comment by SpookyTheFox** [comment]
> 
> 
> > NOPE NOPE NOPE.
> > I watched a test round, this is a horribly unbalanced antag. It gibbs people, breaks machines left and right, moves much faster than normal crew, can vent crawl (which lets it transit the station extremely fast), and revives faster than it can be gibbed by the shuttle, even with people constantly beating on it. For all intents and purposes, unkillable and free to murderbone with zero recourse.
> > It will have to loose ventcrawl at a bare minimum, as well as most of it's damage output, and get a substantial delay in revive time, as 8 seconds is much too short before this could be considered for use.
> 
> Currently has a speed of 2, which is almost the slowest mob we have. Ventcrawl is it's movement and fear factor. Revives were bugged due to Life() not firing. Requires more of a group effort to kill than just "unga bunga assistant with toolbox." With a 2 move speed, anyone can run away. If you had died to it, then you were hunting it. That simple.

I was a ghost observer, not in the round. As an aside, that player did kill one AFK person in engineering, and gib someone else they found dead. When it was on the bridge, and in the operating room, the people being attacked did not appear to be able to get away from it.

On top of high speed ventcrawling, the ability to enter vents rapidly makes it nearly impossible to down in the first place, and having it revive in a few seconds is insulting on top of that.

**Comment by Owai-Seek** [comment]
Considering we're still testing this bobmed schtick, this thing looks -bullshittery- powerful. 30-40 damage per hit? That's automatic wound territory. Gibbing people to heal? This is basically a slaughter demon you can opt into playing.

Even if someones brain/head is dumped on the floor, gibbing someone in maint is basically round removal. Who's going to find it in the dark winding halls?

As for the revive time being too quick, looking in the code, it seems people can just butcher it because it's a simplemob, although I'm unsure how long -that- takes. Meaning you can gut it with combat knives, kitchen knives, cleavers, rapiers, ect. (Different sharp objects have different butcher times.)

I haven't seen the test, and I'm unsure if it's a proper gib, or a shuttlegib on consume, but this seems way too powerful to be outside an event.

**Comment by Feasel** [comment]
@Owai-Seek Wrong. I was using a katana to keep it down as HoS and it would not dissect it.

**Comment by chefferz** [comment]
Also forgot to add this was the objectives of the round.

https://i.imgur.com/meZuGXl.png

Objective #1: Extract 15 compatible genomes.
Objective #2: Kill Feasel, the Head of Security. You only need to kill them once; if they come back, you've still succeeded.
Objective #3: Escape on the shuttle or an escape pod alive and without being in custody.
**Objective #4: Ascend into your true form, and cause as much terror as you can!** 

That last objective is just murderboning. A good balance to this would be making horror form require FULL absorption rather than just DNA stinging.

**Comment by Gandalf2k15** [comment]
> > > NOPE NOPE NOPE.
> > > I watched a test round, this is a horribly unbalanced antag. It gibbs people, breaks machines left and right, moves much faster than normal crew, can vent crawl (which lets it transit the station extremely fast), and revives faster than it can be gibbed by the shuttle, even with people constantly beating on it. For all intents and purposes, unkillable and free to murderbone with zero recourse.
> > > It will have to loose ventcrawl at a bare minimum, as well as most of it's damage output, and get a substantial delay in revive time, as 8 seconds is much too short before this could be considered for use.
> > 
> > 
> > Currently has a speed of 2, which is almost the slowest mob we have. Ventcrawl is it's movement and fear factor. Revives were bugged due to Life() not firing. Requires more of a group effort to kill than just "unga bunga assistant with toolbox." With a 2 move speed, anyone can run away. If you had died to it, then you were hunting it. That simple.
> 
> Six security officers were unable to keep it contained. I watched as the HoS literally killed the ling, put it in the shuttle area to gib it, then it revived not even 8 seconds later to get away, despite missing literally every single organ and part. https://i.imgur.com/gbEszO5.png

It takes the standard ling time to regenerate. However thanks to Citadel's life() changes, it was not regenerating chemicals when dead.

**Comment by Gandalf2k15** [comment]
> > > NOPE NOPE NOPE.
> > > I watched a test round, this is a horribly unbalanced antag. It gibbs people, breaks machines left and right, moves much faster than normal crew, can vent crawl (which lets it transit the station extremely fast), and revives faster than it can be gibbed by the shuttle, even with people constantly beating on it. For all intents and purposes, unkillable and free to murderbone with zero recourse.
> > > It will have to loose ventcrawl at a bare minimum, as well as most of it's damage output, and get a substantial delay in revive time, as 8 seconds is much too short before this could be considered for use.
> > 
> > 
> > Currently has a speed of 2, which is almost the slowest mob we have. Ventcrawl is it's movement and fear factor. Revives were bugged due to Life() not firing. Requires more of a group effort to kill than just "unga bunga assistant with toolbox." With a 2 move speed, anyone can run away. If you had died to it, then you were hunting it. That simple.
> 
> I was a ghost observer, not in the round. As an aside, that player did kill one AFK person in engineering, and gib someone else they found dead. When it was on the bridge, and in the operating room, the people being attacked did not appear to be able to get away from it.
> 
> On top of high speed ventcrawling, the ability to enter vents rapidly makes it nearly impossible to down in the first place, and having it revive in a few seconds is insulting on top of that. Hopefully the life() issue is easily addressed.

If there was AFK killing then it needs to go through the appropriate channels, not github. **Please remember this is in the testing phase.** Which is why it is `test merged.`

**Comment by Gandalf2k15** [comment]
> Considering we're still testing this bobmed schtick, this thing looks -bullshittery- powerful. 30-40 damage per hit? That's automatic wound territory. Gibbing people to heal? This is basically a slaughter demon you can opt into playing.
> 
> Even if someones brain/head is dumped on the floor, gibbing someone in maint is basically round removal. Who's going to find it in the dark winding halls?
> 
> As for the revive time being too quick, looking in the code, it seems people can just butcher it like a simplemob, although I'm unsure how long -that- takes. Meaning you can gut it with combat knives, kitchen knives, cleavers, rapiers, ect. (Different sharp objects have different butcher times.)
> 
> I haven't seen the test, and I'm unsure if it's a proper gib, or a shuttlegib on consume, but this seems way too powerful to be outside an event.

It was slightly bussed due to the fact there was no regenerating chemicals. Until that is addressed they are very easily killed. Beware this is not an easy antag to defeat, so lone-oping it will not work.

**Comment by Gandalf2k15** [comment]
Just as a heads up, I will be addressing the balance issues in due course. Please only leave balance concern comments from here on and out. Thank you.

**Comment by Gandalf2k15** [comment]
> Also forgot to add this was the objectives of the round.
> 
> https://i.imgur.com/meZuGXl.png
> 
> Objective #1: Extract 15 compatible genomes.
> Objective #2: Kill Feasel, the Head of Security. You only need to kill them once; if they come back, you've still succeeded.
> Objective #3: Escape on the shuttle or an escape pod alive and without being in custody.
> **Objective #4: Ascend into your true form, and cause as much terror as you can!**
> 
> That last objective is just murderboning. A good balance to this would be making horror form require FULL absorption rather than just DNA stinging, because it's too easy to just run down a hall and collect stings.

This is actually very useful, thank you. I was contemplating if I should use absorptions instead. I likely will.

**Comment by CameronWoof** [comment]
I was under the impression that Skyrat was a roleplaying server, but the string of high-intensity, low-roleplay antagonist changes recently has made me question that idea.

What roleplay value can we expect from a changeling that eschews its intended stealth/paranoia roleplay in exchange for... more high-powered murder tools and an appearance that makes them automatically valid? I don't think there's much to like here.

**Comment by Gandalf2k15** [comment]
> I was under the impression that Skyrat was a roleplaying server, but the string of high-intensity, low-roleplay antagonist changes recently has made me question that idea.
> 
> What roleplay value can we expect from a changeling that eschews its intended stealth/paranoia roleplay in exchange for... more high-powered murder tools and an appearance that makes them automatically valid? I don't think there's much to like here.
> 
> EDIT: And yes, as Spooky said, ventcrawling needs to go. It's a stupidly powerful ability, particularly for something that benefits from ducking out of combat to regenerate chemicals.

Changelings are supposed to be like "the thing".

**Comment by chefferz** [comment]
I actually did not think about that. Ventcrawling is absolutely way overpowered considering it allows for quick ducking into, and paired with chemical regen allows it to just camp areas while it refills. The sheer size of it also kind of makes no real logical sense ICly. The pipes aren't that big..

**Comment by Gandalf2k15** [comment]
> I actually did not think about that. Ventcrawling is absolutely way overpowered considering it allows for quick ducking into, and paired with chemical regen allows it to just camp areas while it refills. The sheer size of it also kind of makes no real logical sense ICly. The pipes aren't that big.. I'd say nixing the ventcrawling and making it normal speed would compensate for that.

More useful suggestions, thank you. I'll do this.

**Comment by ThatGuyFromThatPlace** [comment]
A changeling shouldn't be able to opt into just saying "Fuck it" to the classic way changelings have been played with stealth, deception, **and proper roleplay** and press the "Be OP murderboner" button.

I have a few suggestions.
One: Absolutely needs to have absorbed people, not stung people... The Thing, used the bodies of the people it absorbed, not, people it brushed against in the hallway. A changeling should have to work up to using its "Ultimate true form" that's a mash together of multiple bodies; Bodies that it can't use if it doesn't have.
Two: It's horror form should have a somewhat short duration. It should be an ability used as a last resort or an emergency panic, not an "Abandon all stealth/RP and rampage" button
Three: Get rid of the **gibbing** people to heal.

But, yeah... This is just, generally really unhealthy for the server... It's a "Make me valid/murderbone license" button, basically, as it is now

**Comment by Somememeguy** [comment]
Why not make it so that it requires a changeling to be absorbed?

**Comment by Gandalf2k15** [comment]
Okay updated with tweaks.

File: modular_skyrat/modules/horrorform/code/modules/mob/hostile/true_changeling.dm [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	T.visible_message("<span class='warning'>[T] begins ripping apart and feasting on [lunch]!</span>", \
+					"<span class='danger'>We begin to feast upon [lunch]...</span>")
+	if(!do_mob(usr, 50, target = lunch))
+		T.devouring = FALSE
+		return 0
+	T.devouring = FALSE
+	if(lunch.getBruteLoss() + lunch.getFireLoss() >= 200) //Overall physical damage, basically
+		T.visible_message("<span class='warning'>[lunch] is completely devoured by [T]!</span>", \
+						"<span class='danger'>You completely devour [lunch]!</span>")
+		lunch.gib()
```
Comment (NotRanged) [inline modular_skyrat/modules/horrorform/code/modules/mob/hostile/true_changeling.dm:189]: Consider removing this part, permanent round removal bad

Comment (Gandalf2k15) [reply modular_skyrat/modules/horrorform/code/modules/mob/hostile/true_changeling.dm:189]: This is intended as a balance feature. And generally speaking providing the head remains they shouldn't be permanently round removed.

**Comment by Gandalf2k15** [comment]
Removes the gib

**Pull request merged**